bee-crypto/src/ternary/hash.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use bee_ternary::{Trits, T1B1};
+
+use std::{cmp::PartialEq, fmt, hash};
+
+/// The length of a hash in units of balanced trits.
+pub const HASH_LENGTH: usize = 243;
+
+/// Ternary cryptographic hash.
+#[derive(Copy, Clone)]
+pub struct Hash([i8; HASH_LENGTH]);
+
+impl Hash {
+    /// Creates a new `Hash` from trits.
+    pub fn new(trits: [i8; HASH_LENGTH]) -> Self {
+        Self(trits)
+    }
+
+    /// Creates a hash filled with zeros.
+    pub fn zeros() -> Self {
+        Self([0; HASH_LENGTH])
+    }
+
+    /// Reinterprets the `Hash` as `Trits`.
+    pub fn as_trits(&self) -> &Trits<T1B1> {
```

Comment:
Is `as_trits` the only route through which the inner trits are accessed? I've been meaning to create a static trit encoding that wraps a `[i8; N]`. I haven't done it yet since there hasn't been call for it, but this seems like a good candidate.

Reply:
It's being done (by you)